Chatter/AvatarPicker/AvatarPickerRootView.swift

```diff
@@ -0,0 +1,135 @@
+public class AvatarPickerRootView: UIView {
+
+    typealias AvatarPickerDismissHandler = () -> Void
+    typealias AvatarTypeChangeHandler = (AvatarType) -> Void
+
+    var delegate: UICollectionViewDelegate? {
+        set {
+            collectionView.delegate = newValue
+        }
+        get {
+            return nil
+        }
+    }
+    var dataSource: UICollectionViewDataSource? {
+        set {
+            collectionView.dataSource = newValue
+        }
+        get {
+            return nil
+        }
+    }
+
+    var dismissHandler: AvatarPickerDismissHandler?
+    var avatarTypeChangeHandler: AvatarTypeChangeHandler?
+
+    private let backButton: UIButton = {
+        let backButton = UIButton(type: .custom)
+        backButton.setImage(#imageLiteral(resourceName: "smackBack"), for: .normal)
+        backButton.heightAnchor
+            .constraint(equalToConstant: 30)
+            .isActive = true
+        backButton.widthAnchor
+            .constraint(equalToConstant: 30)
+            .isActive = true
+        return backButton
+    }()
+
+    private let segmentedControl: UISegmentedControl = {
+        let segmentedControl = UISegmentedControl(items: ["Dark", "Light"])
+        segmentedControl.selectedSegmentIndex = 0
+        segmentedControl.heightAnchor
+            .constraint(equalToConstant: 30)
+            .isActive = true
+        segmentedControl.widthAnchor
+            .constraint(equalToConstant: 150)
+            .isActive = true
+        return segmentedControl
+    }()
+
+    private let collectionView: UICollectionView = {
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .vertical
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        collectionView.backgroundColor = UIColor.white
+        collectionView.contentInset = UIEdgeInsets(top: 0, left: 20, bottom: 0, right: 20)
+        collectionView.register(AvatarPickerCell.self, forCellWithReuseIdentifier: "avatarCell")
+        return collectionView
+    }()
+
+    init() {
+        super.init(frame: .zero)
+        backButton.addTarget(nil, action: #selector(dismiss), for: .touchUpInside)
+        segmentedControl.addTarget(nil, action: #selector(update), for: .valueChanged)
+    }
+
+    @available(*, unavailable, message: "Use init() instead")
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    public override func didMoveToWindow() {
+        backgroundColor = UIColor.white
+        constructHierarchy()
+        activateConstraints()
+    }
+
+    func reloadData() {
+        collectionView.reloadData()
+    }
+
+    @objc private func dismiss() {
+        dismissHandler?()
+    }
+
+    @objc private func update(_ selector: UISegmentedControl) {
+        if selector.selectedSegmentIndex == 0 {
+            avatarTypeChangeHandler?(.dark)
+        } else {
+            avatarTypeChangeHandler?(.light)
```

Comment:
In the case of changing the number of items all. other than one will produce light avatar. Consider using a switch with a default value.